bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,451 @@
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.Background;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.Color;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.ColorMode;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.Gamma;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.Luminance;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.PowerStatus;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.Source;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorDevice.Switch;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EpsonProjectorHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pauli Anttila, Yannick Schaus - Initial contribution
+ */
+@NonNullByDefault
+public class EpsonProjectorHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorHandler.class);
+
+    private @Nullable EpsonProjectorConfiguration config;
+
+    private @Nullable EpsonProjectorDevice device;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public EpsonProjectorHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channelId);
+            sendDataToDevice(epsonCommand, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(EpsonProjectorConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            if (StringUtils.isNotEmpty(config.serialPort)) {
+                device = new EpsonProjectorDevice(config.serialPort);
+            } else if (StringUtils.isNotEmpty(config.host) && config.port > 0) {
+                device = new EpsonProjectorDevice(config.host, config.port);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                return;
+            }
+
+            try {
+                device.connect();
+                updateStatus(ThingStatus.ONLINE);
+
+                int pollingInterval = config.pollingInterval;
+                if (pollingInterval == 0) {
+                    pollingInterval = 10000;
+                }
+
+                List<Channel> channels = this.thing.getChannels();
+
+                pollingJob = scheduler.scheduleAtFixedRate(new Runnable() {
```

Comment:
Please use a lambda and maybe put the body in a method and call it with a method reference